scipy/signal/lfilter.c.src

```diff
@@ -63,6 +64,62 @@ RawFilter(const PyArrayObject * b, const PyArrayObject * a,
           const PyArrayObject * zf, PyArrayObject * y, int axis,
           BasicFilterFunction * filter_func);
 
+PyObject*
+convert_shape_to_errmsg(intp ndim, intp *Xshape, intp *Vishape, intp theaxis, intp val)
+{
+    intp j, expect_size;
+    PyObject *msg, *tmp, *msg1, *tmp1;
+
+    if (ndim == 1) {
+        msg = PyUString_FromFormat("Unexpected shape for zi: expected (%"   \
+                                    NPY_INTP_FMT ",), found (%" NPY_INTP_FMT \
+                                    ",).", val, Vishape[0]);
+        return msg;
+    }
+
+    msg = PyUString_FromString("Unexpected shape for zi:  expected (");
+    if (!msg) {
+        return 0;
+    }
+
+    msg1 = PyUString_FromString("), found (");
+    if (!msg1) {
+        Py_DECREF(msg);
+        return 0;
+    }
+
+    for (j = 0; j < ndim; ++j) {
+        expect_size = j != theaxis ? Xshape[j] : val;
+
+        if (j == ndim - 1) {
+            tmp = PyUString_FromFormat("%" NPY_INTP_FMT, expect_size);
+            tmp1 = PyUString_FromFormat("%" NPY_INTP_FMT, Vishape[j]);
+        } else {
+            tmp = PyUString_FromFormat("%" NPY_INTP_FMT ",", expect_size);
+            tmp1 = PyUString_FromFormat("%" NPY_INTP_FMT ",", Vishape[j]);
+        }
+        if (!tmp) {
+            Py_DECREF(msg);
+            Py_XDECREF(tmp1);
+            return 0;
+        }
+        if (!tmp1) {
+            Py_DECREF(msg);
+            return 0;
+        }
```

Comment:
Missing ``Py_DECREF(msg1)`` in the above two if-statements, and  ``Py_DECREF(tmp)`` in the second one.